src/main/java/de/focusshift/zeiterfassung/overtime/OvertimeServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package de.focusshift.zeiterfassung.overtime;
+
+import de.focusshift.zeiterfassung.report.ReportDay;
+import de.focusshift.zeiterfassung.report.ReportServiceRaw;
+import de.focusshift.zeiterfassung.user.UserIdComposite;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+
+@Service
+class OvertimeServiceImpl implements OvertimeService {
+
+    private final ReportServiceRaw reportServiceRaw;
+
+    OvertimeServiceImpl(ReportServiceRaw reportServiceRaw) {
+        this.reportServiceRaw = reportServiceRaw;
+    }
+
+    @Override
+    public Map<UserIdComposite, OvertimeHours> getOvertimeForDate(LocalDate date) {
+
+        final ReportDay reportDay = reportServiceRaw.getReportDayForAllUsers(date);
+
+        return reportDay.deltaDurationByUser().entrySet().stream().collect(
+            toMap(
+                Map.Entry::getKey,
+                entry -> new OvertimeHours(entry.getValue().duration())
```

Comment:
Do we need OvertimeHours with Duration zero? Should we filter them?

Reply:
die methode sollte denke ich für jeden user einen eintrag haben, damit kein "key does not exist" oder wie es heißt passiert zur Laufzeit.

die frage "hat user überstunden" muss ja eh immer gestellt werden beim Konsumenten. und ob das ein `hasKey` oder ein `is Duration.ZERO` ist. ist egal, oder?

Reply:
habe im rabbit publisher die Prüfung eingebaut. Zero Überstunden werden nicht auf den rapid publiziert.